### PR TITLE
lnd: improve service initialization

### PR DESCRIPTION
**Pull request by erikarvstedt**
#### Copy of commit msg

- Use systemd startup notification.
- Use new option `--wallet-unlock-password-file` instead of manual unlocking.

@Kixunil, this is also relevant for cryptoanarchy-deb-repo-builder.

### CODE REVIEW

**Comment by jonasnick** [comment]
This fixes #377

**Review by jonasnick — APPROVED** [review]
This is really nice.

ACK a984f3cf6f7fc2147e05f5b58c83a09cb31f5416

I've modified #377 a bit so it can be left open for the initwallet tool.

File: modules/lnd.nix [hunk 1]
```diff
@@ -207,31 +207,27 @@ in {
...
-            attempts=250
-            while ! {
-              exec 3>/dev/tcp/${nbLib.address cfg.restAddress}/${toString cfg.restPort} && exec 3>&-
-            } &>/dev/null; do
-              ((attempts-- == 0)) && { echo "lnd REST service unreachable"; exit 1; }
-              sleep 0.1
-            done
-
             if [[ ! -f ${networkDir}/wallet.db ]]; then
+
```
[the remark this answers is not in the document]
Comment (erikarvstedt) [reply modules/lnd.nix:230]: I had added it for readability, but it's inconsistent with our overall formatting style, so I've removed it.

**Comment by Kixunil** [comment]
@erikarvstedt I know, I was the one who made the systemd PR and bothered LND devs to add password file unlock option. ;)
It's already in experimental.

**Comment by erikarvstedt** [comment]
@Kixunil Ah, I missed this when skimming `master`. You might want to use the status polling approach of this PR instead of [sleeping for a fixed duration](https://github.com/debian-cryptoanarchy/cryptoanarchy-deb-repo-builder/blob/b0fc61da284a96a8f547cf2dc6b10f7646f737d7/lnd/auto_unlock.sh#L27) when waiting for lnd server startup.

**Comment by Kixunil** [comment]
I'm not sure what that status method returns when the wallet is unlocked but not fully initialized. Anyway I also plan to rewrite it to use lndinit and I happen to also be the person who inspired it. :) I suspect my time will be better spent getting `lndinit` to be tested and released than writing code that should be removed soon.

**Comment by erikarvstedt** [comment]
Note to reviewers: At the point where this PR fetches the service status, the text is guaranteed to be either `Wallet locked` or `Wallet unlocked`.

**Review by nixbitcoin — APPROVED** [review]
utACK d8af0aca02dd846d66b9528bfb36764cc96c78bc

Can somebody using lnd in production confirm this is working for them @mmilata?

**Comment by mmilata** [comment]
Nice! I'll give it a try in the next few days.

**Review by mmilata — APPROVED** [review]
Works!

**Review by jonasnick — COMMENTED** [review]
Thanks @mmilata 

ACK d8af0aca02dd846d66b9528bfb36764cc96c78bc

**Pull request merged**